resources/prosody-plugins/mod_auth_token.lua

```diff
@@ -74,18 +74,28 @@ function provider.delete_user(username)
 	return nil;
 end
 
+local function validate_result(session, res, error, reason)
```

Comment:
What is the purpose of this function?  I believe we'll want a different message based on if the failure is happening in key fetch, token validation or in post auth.  So it's better to simply handle each of these separately.